src/helpers/git_helpers/remote_helpers.sh

```diff
@@ -23,7 +23,7 @@ function remote_domain {
 # Returns the USER/REPO for the remote repository
 function remote_repository_name {
   local domain=$(remote_domain)
-  remote_url | sed -E "s#.*$domain[/:](.+)#\1#" | sed "s/\.git$//"
+  remote_url | sed -E "s#.*$domain[/:](.+)#\1#" | sed "s/\.git$//" | sed 's#ssh://git@[^/]*/##'
```

Comment:
I believe the updated tests will pass even without this change. The new example remote urls looks very similar in structure to the `http://` and `https://` versions and thus I don't think any logic change is needed to support them.

I also ran the following test in my terminal and doesn't appear the change needed.
```sh
echo "ssh://git@github.com/Originate/git-town" | sed -E "s#.*github.com[/:](.+)#\1#" 
# outputs Originate/git-town
```